packages/backend/src/routes/searchForId.mjs

```diff
@@ -0,0 +1,48 @@
+import catchError from '../helpers/catchError.mjs'
+import { TimestampLoader } from '../helpers/timestampLoader.mjs'
+
+export default ({ app, db, synchronizer }) => {
+  const handler = async (req, res) => {
+    const { id } = req.params
+    const deployment = await db.findOne('Attester', {
+      where: {
+        _id: id,
+      },
+    })
+
+    const attestations = await db.findMany('Attestation', {
+      where: {
+        epochKey: id,
+      },
+    })
+    const epochKeyItems = await TimestampLoader.inject(attestations, db)
+
+    const signUps = await db.findMany('UserSignUp', {
+      where: {
+        commitment: id,
+      },
+    })
+    const userItems = await TimestampLoader.inject(signUps, db)
+
+    let type = ''
```

Comment:
Instead of looking in all collections you should check one at a time, and return early if you find an entry. For example, if you find an attester for the `id` there's no need to search the database for signups/attestations.

Something like this

```js
const deployment = await db.findOne('Attester', {
      where: {
        _id: id,
      },
    })

if (deployment) {
  res.json({ type: 'attester', data: deployment })
  return
}
```